server/lib/scheduler/Scheduler.ts

```diff
@@ -251,24 +222,19 @@ class Scheduler {
       );
 
       const oracleTrees = await filter(highestPriorityTrees, t => {
-        // TODO: make sure fetchTrees.. is properly getting the oracle relations array
         return t.oracleRelations.length > 0;
       });
 
       let oracleTreesToConsider = oracleTrees;
 
       while (oracleTreesToConsider.length > 0) {
-        const rootWorkspacesOfTrees: Workspace[] = oracleTreesToConsider.map(
-          t => t.rootWorkspace,
-        );
-        const leastRecentlyWorkedOnTreeRootWorkspaces: Workspace[] = await this.getTreesWorkedOnLeastRecentlyByUser(
+        const randomlySelectedRootWorkspace = this.getRandomTreeRootWorkspaceWorkedOnLeastRecentlyByUser(
           userId,
-          rootWorkspacesOfTrees,
+          oracleTreesToConsider,
         );
-        const randomlySelectedRootWorkspace = pickRandomItemFromArray(
-          leastRecentlyWorkedOnTreeRootWorkspaces,
-        );
-
+        if (!randomlySelectedRootWorkspace) {
```

Comment:
Maybe throw an error here? This condition should never be true so if it is something went wrong. Same thing with line 285.

Reply:
On second look, definitely agreed. We'd want to know if this is happening, instead of people just getting fewer results.